### ISSUE TITLE
Fix Renovate config

### ISSUE THREAD

**Issue by rarkins** [tdeekens/flopflip#120]
Hi, I noticed some problems in your config and wanted to discuss them before submitting a Pull Request, as I'm not 100% sure I know how you'd like it. If you think the documentation is at fault, please feel free to point that out too.

First up, I want to alert you to https://github.com/renovateapp/renovate/pull/1419
It's currently a design feature - but essentially a limitation - that config in a `package.json` applies only to *that* `package.json`. Early on it seemed a good way to provide different config to different package files in a monorepo, but I intend to change it. Because it's a breaking change, I was wanting to wait for feedback before merging it. In the meantime, you can move your config back to any of these files: `renovate.json`, `.renovaterc`, or `.renovaterc.json` so that the config applies to the entire monorepo.

If the config *is* placed inside a `package.json` once 1419 is merged then it actually needs to be inside a `renovate` object, not `renovate-config`. The latter is used for Renovate presets (that function like eslint shared configs) and not for cases like this. So in fact.. none of your config is activated right now - this repo is getting the app's default settings, which fortunately are sane.

Reviewing the content of your config:

```
      "extends": [
        "config:base"
      ],
```
This is good. Over time the bot's default settings are getting less "opinionated", as I pushed a few of those opinions into the `config:base` preset, which is then recommended when onboarding. For example with this preset it will automatically group dependencies that come from known monorepos like babel, react, etc.

```
      "packagePatterns": [
        "^@flopflip/"
      ],
```

This is technically invalid config, because `packagePatterns` needs to be inside a "packageRule", so it just ends up doing nothing. If you tell me what you were hoping to achieve, I can tell you how to do it.

```
      "lockFileMaintenance": {
        "enabled": true
      },
```

This is fine. It means every monday morning Renovate will regenerate your `yarn.lock` from scratch and you should see a "lock file maintenance" PR if there are any changes since the last time it was updated.

```
      "semanticCommits": false,
```

It looks like you are using semantic commits in this repo - did you definitely want to turn them off? You can configure both commit scope and type if desired.

```
      "prTitle": "{{semanticCommitType}}({{semanticCommitScope}}): {{#if isPin}}Pin{{else}}{{#if isRollback}}Roll back{{else}}Update{{/if}}{{/if}} {{depName}} to {{#unless isRange}}v{{/unless}}{{#if isMajor}}{{newVersionMajor}}.x{{else}}{{newVersion}}{{/if}}",
      "commitMessage": "{{semanticCommitType}}({{semanticCommitScope}}): Update {{depName}} to {{#unless isRange}}v{{/unless}}{{newVersion}}",
```

I'm not sure you need to configure/override these?

```
      "packageRules": [
        {
          "packagePatterns": ["*"],
          "excludePackagePatterns": [],
          "enabled": false
        }
      ]
```

This will in effect match every package name and therefore disable Renovate for all packages. Did you want to turn off all package updating except for your `@flopflip` ones or something like that? If so then the easiest thing is to add the `^@flopflip/` regex to `excludePackagePatterns`.

FYI I'll be back online tomorrow to follow up on any response.

**Comment by tdeekens** [comment]
Thanks you so much for alerting me about the mistakes I made [here](https://github.com/tdeekens/flopflip/pull/121). To be honest I might just not have reflected enough when adding things. I mostly went with what I found over [here](https://github.com/seek-oss/renovate-config-seek). Maybe something worth checking if things there contain similar problems.

I fixed up things in this PR and merged it in already. 

Regarding your comments

1. `packagePatterns`: Removed
2. `lockFileMaintenance`: Intended so I kept it
3. `semanticCommits` & messages: Opted back into the defaults
4. `packageRules`: My bad. I thought of disabling it for the `demo` but I don't yet have a good reason to so I removed it too.

This makes the config quite a bit shorter which is nice. Thanks again for keeping an eye on thing!
